### PR TITLE
build: update devcontainer

### PR DESCRIPTION
**Pull request by chemelli74**

### CODE REVIEW

**Comment by rokam** [comment]
@ chemelli74 I suggest that we add a matrix of home-assistant versions. That way mypy could check for every version after 2023.8.0. What do you think?

**Comment by chemelli74** [comment]
I like the idea

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# [Choice] Python version (use -bullseye variants on local arm64/Apple Silicon): 3, 3.10, 3.9, 3.8, 3.7, 3.6, 3-bullseye, 3.10-bullseye, 3.9-bullseye, 3.8-bullseye, 3.7-bullseye, 3.6-bullseye, 3-buster, 3.10-buster, 3.9-buster, 3.8-buster, 3.7-buster, 3.6-buster
+FROM mcr.microsoft.com/vscode/devcontainers/python:${VARIANT}
```
Comment (wuwentao) [inline .devcontainer/Dockerfile:2]: missed `ARG VARIANT=3.11-bullseye`

Comment (wuwentao) [reply .devcontainer/Dockerfile:2]: sorry, found it exist in `devcontainer.json` file

**Review by wuwentao — APPROVED** [review]

**Pull request merged**